Enlinkd Topology Map Layers Documentation (#4862)
OpenNMS/opennms: docs/modules/operation/pages/topology/enlinkd/introduction.adoc

[[ga-enlinkd-layers]]
== Topology Layers

The Enhanced Linkd topology map supports layered protocol maps.

Choose menu:Maps[Topology] to display the topology map, and click the Layers icon on the right panel of the map to access available layers:

* *All* (default) – displays the topology information Enhanced Linkd discovers, for all supported protocols (OSPF, ISIS, Bridge, LLDP, and CDP).
It also includes a user-defined option.

* *Cdp* – displays the Cisco Discovery Protocol topology information Enhanced Linkd discovers.

* *Lldp* – displays the Link Layer Discovery Protocol topology information Enhanced Linkd discovers.

* *Layer2* – displays CDP and LLDP combined topology information Enhanced Linkd discovers.

* *Bridge* – shows only Bridge links.

* *Ospf* – displays the Open Shortest Path First (OSPF) topology information Enhanced Linkd discovers.

* *Isis* – displays the IS-IS topology information Enhanced Linkd discovers.

*Layer3* – displays OSPF and IS-IS combined topology information Enhanced Linkd discovers.

> *UserDefined* – displays user-defined topology information.

Reviewer: ```suggestion
* *UserDefined* – displays user-defined topology information.
```

Additionally, the supported link discovery implementations are described as well as a list of the SNMP MIBs that the SNMP agents must expose for EnLinkd to gather links between nodes.
You can view detailed information about a node's connections (discovered links) and supporting link data on the node detail page within the {page-component-title} Web UI.

[[ga-enlinkd-layers]]
== Topology Layers

> Enhanced Linkd Topology Map supports Layers protocol maps.

Reviewer: ```suggestion
The Enhanced Linkd topology map supports layered protocol maps.
```

Reviewer: Could we add a reason why having the layers is useful?

Author: Well, this is a good point.

Topology Maps is a compelling point.

The idea behind having protocols maps is that you can have a better view removing some data.

LLDP and CDP are very similar protocols. By the way you can decide to look only on CDP network and only on LLDP network. This can give you an idea if the protocol is working properly and if is or not enabled on all devices.

OSPF maps are different, because you can be interested in inter routing connection.

Layers can be useful in many cases. You can view the network from different point of view and is what layers give you.

I know is generic but, in terms of use cases one could be:

I'm not interested in OSPF and IS-IS Link and I want just a map of Layer two links.
I'm only interested in looking at Layer 3 topology, so I do not want to see Layer 2 related links.

I'm only interested in bridge topology.

> Click the Layers Button on the right panel of the Topology Map to access available Layers.

Reviewer: ```suggestion
Choose menu:Maps[Topology] to display the topology map, and click the Layers icon on the right panel of the map to access available layers:
```

> Supported Layers Maps:

Reviewer: ```suggestion

```

> *All* _(Selected by default)_ displays the topology information discovered by the Enhanced Linkd,

Reviewer: ```suggestion
* *All* (default) – displays the topology information Enhanced Linkd discovers, for all supported protocols (OSPF, ISIS, Bridge, LLDP, and CDP).
It also includes a user-defined option. 
```
These should be a bulleted list. 
Is the user-defined option to customize?

> *All* _(Selected by default)_ displays the topology information discovered by the Enhanced Linkd,

Reviewer: ```suggestion
* *All* (default) – displays the topology information Enhanced Linkd discovers, for all supported protocols (OSPF, ISIS, Bridge, LLDP, and CDP).
It also includes a user-defined option. 
```
These should be a bulleted list. 
Is the user-defined option to customize?

Author: Yup, seems reasonable.

for all supported protocols OSPF, ISIS, Bridge, LLDP and CDP including User Defined to generate the overall topology

> *CDP* displays the Cisco Discovery Protocol topology information discovered by the Enhanced Linkd

Reviewer: ```suggestion
* *CDP* – displays the Cisco Discovery Protocol topology information Enhanced Linkd discovers.
```

Reviewer: Out of scope for this PR, but it would be nice if the labels in the UI were displayed correctly ... CDP instead of Cdp, UDP instead of Udp.

Reviewer: Note also that UDP is missing from this list. Please add.

> *LLDP* displays the Link Layer Discovery Protocol topology information discovered by the Enhanced Linkd

Reviewer: ```suggestion
* *LLDP* – displays the Link Layer Discovery Protocol topology information Enhanced Linkd discovers.
```

> *Layer2* displays CDP and LLDP combined topology information discovered by Enhanced Linkd

Reviewer: ```suggestion
* *Layer2* – displays CDP and LLDP combined topology information Enhanced Linkd discovers.
```

> *Bridge* (Only Bridge Links are shown)

Reviewer: ```suggestion
* *Bridge* – shows only Bridge links.
```

> *OSPF* displays the OSPF topology information discovered by the Enhanced Linkd

Reviewer: ```suggestion
* *OSPF* – displays the Open Shortest Path First (OSPF) topology information Enhanced Linkd discovers.
```

> *ISIS* displays the IS-IS topology information discovered by the Enhanced Linkd

Reviewer: ```suggestion
* *ISIS* – displays the IS-IS topology information Enhanced Linkd discovers.
```

> *Layer3* displays OSPF and IS-IS combined topology information discovered by Enhanced Linkd

Reviewer: ```suggestion
*Layer3* – displays OSPF and IS-IS combined topology information Enhanced Linkd discovers.
```

> *UserDefined* displays the User Defined topology information

Reviewer: ```suggestion
*UserDefined* – displays user-defined topology information.
```
